packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarousel/Carousel/useCarouselCollection.ts

```diff
@@ -0,0 +1,129 @@
+import { Context, ContextSelector, createContext, useContextSelector } from '@fluentui/react-context-selector';
+import { CarouselPageChangeData, CarouselStore } from './Carousel.types';
+import * as React from 'react';
+import { EventHandler } from '@fluentui/react-utilities';
+
+/**
+ * A hook for managing a collection of carousel pages
+ */
+export const createCarouselStore = (): CarouselStore => {
+  let values: string[] = [];
+  let listeners: Array<() => void> = [];
+
+  const carouselStore = {
+    addValue(value: string) {
+      values = [...values, value];
+
+      emitChange();
+    },
+    insertValue(value: string, prev: string | null) {
+      if (!prev) {
+        values = [value, ...values];
+      } else {
+        const pos = values.indexOf(prev);
+        values.splice(pos + 1, 0, value);
+        // Required to be defined as a 'new' array for useSyncExternalStore
+        values = [...values];
+      }
+      emitChange();
+    },
+    removeValue(value: string) {
+      const pos = values.indexOf(value);
+      values.splice(pos, 1);
+      // Required to be defined as a 'new' array for useSyncExternalStore
+      values = [...values];
+      emitChange();
+    },
+    subscribe(listener: () => void) {
+      listeners = [...listeners, listener];
+
+      return () => {
+        listeners = listeners.filter(l => l !== listener);
+      };
+    },
+    getSnapshot() {
+      return values;
+    },
+    getIndex(index: number) {
+      if (index < values.length) {
+        return values[index];
+      }
+      // Safety fallback if not found, undefined is ok too.
+      return values[0];
+    },
+  };
+
+  function emitChange() {
+    for (const listener of listeners) {
+      listener();
+    }
+  }
+
+  return {
+    ...carouselStore,
+  };
+};
+
+export const useCarouselCollection_unstable = () => {
+  const [store] = React.useState(() => createCarouselStore());
+  const [value, _setValue] = React.useState<string | null>(null);
```

Comment:
This should use `useControllableState()` and accept `value` from arguments of `useCarouselCollection_unstable()` i.e.

```tsx
export const useCarouselCollection_unstable = (options: { value }) => {
  const [value, _setValue] = useControllableState<string | null>({
    value: options.value,
    defaultValue: options.defaultValue,
    initialState: '',
  });
```